http://sourceware.org/ml/gdb-patches/2016-04/msg00181.html
Subject: [testsuite patch] Fix 7.11 regression: gdb.dwarf2/dw2-undefined-ret-addr.exp [now with the patch]


--oyUTqETQ0mS9luUI
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline

Hi,

a038fa3e14a477d4d72a26c2e139fa47d2774be2 is the first bad commit
commit a038fa3e14a477d4d72a26c2e139fa47d2774be2
Author: Markus Metzger <markus.t.metzger@intel.com>
    stack: check frame_unwind_caller_id
[PATCH v3 2/3] stack: check frame_unwind_caller_id
https://sourceware.org/ml/gdb-patches/2016-02/msg00363.html
Message-Id: <1455207283-12660-2-git-send-email-markus.t.metzger@intel.com>

 (gdb) PASS: gdb.dwarf2/dw2-undefined-ret-addr.exp: up refuses to go up
 info frame
 Stack level 0, frame at 0x0:
- rip = 0x40059a in stop_frame (dw2-undefined-ret-addr.c:22); saved rip = <not saved>
+ rip = 0x40059a in stop_frame (dw2-undefined-ret-addr.c:22); saved rip = <unavailable>
  Outermost frame: outermost
  source language c.
  Arglist at 0x7fffffffd610, args: 
  Locals at 0x7fffffffd610, Previous frame's sp is 0x7fffffffd620
  Saved registers:
   rbp at 0x7fffffffd610
-(gdb) PASS: gdb.dwarf2/dw2-undefined-ret-addr.exp: info frame
+(gdb) FAIL: gdb.dwarf2/dw2-undefined-ret-addr.exp: info frame

It happens on normal native Fedora 23 x86_64.

I am not completely sure about the reasons but it looks to me GDB behavior has
just changed.

OK for check-in?


Thanks,
Jan

--oyUTqETQ0mS9luUI
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline; filename=1

gdb/testsuite/ChangeLog
2016-04-08  Jan Kratochvil  <jan.kratochvil@redhat.com>

	* gdb.dwarf2/dw2-undefined-ret-addr.exp: Expect <unavailable>.

diff --git a/gdb/testsuite/gdb.dwarf2/dw2-undefined-ret-addr.exp b/gdb/testsuite/gdb.dwarf2/dw2-undefined-ret-addr.exp
index c666584..e43054b 100644
--- a/gdb/testsuite/gdb.dwarf2/dw2-undefined-ret-addr.exp
+++ b/gdb/testsuite/gdb.dwarf2/dw2-undefined-ret-addr.exp
@@ -48,7 +48,7 @@ gdb_test "up" \
 # doesn't cause an error, and shows "<not saved>".
 gdb_test "info frame" [multi_line \
     "Stack level 0, frame at $hex\:" \
-    " rip = $hex in stop_frame \\(dw2-undefined-ret-addr\\.c:22\\); saved rip = <not saved>" \
+    " rip = $hex in stop_frame \\(dw2-undefined-ret-addr\\.c:22\\); saved rip = <unavailable>" \
     " Outermost frame: outermost" \
     " source language c\\." \
     " Arglist at $hex, args\: " \

--oyUTqETQ0mS9luUI--

